cmux-tui/bindings/python/cmux/client_defaults.py

```
@@ -1,18 +1,72 @@
-    runtime = os.environ.get("XDG_RUNTIME_DIR")
-    if not runtime:
-        runtime = os.environ.get("TMPDIR") or "/tmp"
-    return os.path.join(runtime, f"cmux-tui-{os.getuid()}", f"{session}.sock")
+    """Return the default socket path after validating the session name."""
+    validate_session_name(session)
+    runtime = _runtime_base()
+    file_name = f"{session}.sock"
+    preferred = os.path.join(runtime, f"cmux-tui-{os.getuid()}", file_name)
+    if _unix_socket_path_fits(preferred):
+        return preferred
+    fallback = os.path.join("/tmp", f"cmux-tui-{os.getuid()}", file_name)
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)